backend/routes/questions.js

```diff
@@ -1,88 +1,43 @@
-const router = require("express").Router();
-const Question = require("../models/question.model");
-const Answer = require("../models/answer.model");
+const router = require('express').Router();
+const Question = require('../models/question.model');
+const Answer = require('../models/answer.model');
 
 // all the questions:
-router.route("/").get((req, res) => {
-  Question.find()
-    .populate("answers")
-    .select("answers")
-    .then((Find, err) => {
-      if (Find) {
-        res.status(200).send(Find);
-      } else {
-        res.status(400).send("No valid entry found for Data :" + err);
-      }
-    });
+router.route('/').get((req, res) => {
+	Question.find()
+		.populate('answer')
+		.then((questions) => res.json(questions))
+		.catch((err) => res.status(400).json('Error:' + err));
 });
 
-//create new Question :
-router.route("/ask").post((req, res) => {
-  const newQuestion = new Question(req.body.question);
-  newQuestion.save((err, question) => {
-    if (err) {
-      return res.status(500).json({ "something is wrong": err });
-    } else {
-      return res.status(201).json(question);
-    }
-  });
+// create new Question :
+router.route('/ask').post((req, res) => {
+	const newQuestion = new Question(req.body);
+	newQuestion.save().then((result) => res.json(result)).catch((err) => res.status(500).json({ err }));
 });
 
 // Get Question By ID
-router.route("/:questionId").get((req, res) => {
-  const questionId = req.params.questionId;
-  Question.findById(req.params.questionId, (err, questionWithId) => {
-    if (questionWithId) {
-      res.status(200).json(questionWithId);
-    } else {
-      res.status(400).json("No valid entry found for ID ");
-    }
-  }).catch((err) => {
-    res.status(500).json({ error: err });
-  });
+router.route('/:questionId').get((req, res) => {
+	Question.findById(req.params.questionId)
+		.then((question) => res.json(question))
+		.catch((err) => res.status(400).json({ err }));
 });
 
-//Answer part:
-router.route("/:id/answer").post((req, res) => {
-  Question.findById(req.params.id, (err, question) => {
-    console.log(question);
-    const newAnswer = new Answer(req.body.answers);
-    newAnswer.save((err, answer) => {
-      if (err) {
-        res.status(500).json({ "something is wrong ": err });
-      }
-      question.answers.push(answer);
-      newAnswer.save((err, newAnswer) => {
-        if (err) {
-          return res.status(500).json({ "something is wrong sec ": err });
-        } else {
-          return res
-            .status(201)
-            .json({ question: question, newAnswer: newAnswer });
-        }
-      });
-    });
-  });
+// Answer part:
+router.route('/:id/answer').post((req, res) => {
```

Comment:
Hi @Hamidreza-Fakharzadeh - I think if you want to use this chained `.then` approach, then you will have to make sure you are returning a promise inside each `.then` block. Are you sure this working? Is it definitely saving the answer in your database and returning it as an answer to the question when you go to show route for the question? Thanks